.github/workflows/clang_lint/clang_lint/custom_clang_format_commenter.py

```diff
@@ -0,0 +1,62 @@
+# /usr/bin/python
+""" Exports a function to run clang-format against the non-ignored
+sources and turn the generated diff into review comments, if any. """
+#
+# Copyright (C) 2021 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+
+
+import subprocess
+import git
+
+from clang_lint.diff_to_review import diff_to_review
+from clang_lint.clang_lint import files_to_lint
+
+
+def review_comments(repository_root: str, pull_request_id: int,
+                    review: str = "REQUEST_CHANGES",
+                    fallback_style: str = "LLVM"):
+    project_dir = "DistroLauncher"
+    sources = files_to_lint.files_to_lint(repository_root, project_dir)
+    proc = subprocess.run(["clang-format", "--fallback-style",
+                           fallback_style, "-i", "--style", "file"] +
+                          sources)
+    if proc.returncode != 0:
+        print("Failed to run clang-format.")
+        return None
+
+    repo = git.Repo(repository_root)
```

Comment:
Good point. This piece of the workflow (as is) totally depends on Git. Thus wrapping in a try-except clause and failing fast makes sense.
It would be possible to make a copy of the work directory before running the formatter and running the Python equivalent to `git diff --no-index dirA dirB`, which obviously requires detecting first whether we are in a git repository or not. Do you think it is worthy? For now I'm just pushing the try-except.

Reply:
The try-except is good enough IMHO for now. At least, if we install it on a non git repo, we will see it quick and then, decide if the copy + manual diff is better or not. Thanks!